torch/_inductor/compile_fx.py

```diff
@@ -770,321 +770,356 @@ def _compile_fx_inner(
     return compiled_graph
 
 
-def fx_codegen_and_compile(
-    gm: GraphModule,
-    example_inputs: Sequence[InputType],
-    # This is derivable from the other inputs to this function, but we pass it
-    # in explicitly because it's nontrivial to compute
-    inputs_to_check: Sequence[int],
-    **graph_kwargs: Unpack[_CompileFxKwargs],
-) -> OutputCode:
-    # Sorry about the mess, we need graph_kwargs to continue to be able
-    # to propagate it further on
-    # TODO: _CompileFxKwargs actually has stronger types than in the
-    # signature, need to tighten it up
-    assert "cudagraphs" in graph_kwargs and graph_kwargs["cudagraphs"] is not None
-    cudagraphs: BoxedBool = graph_kwargs["cudagraphs"]
-    static_input_idxs: Sequence[int] = graph_kwargs.get("static_input_idxs", ())
-    is_backward: bool = graph_kwargs.get("is_backward", False)
-    graph_id: Optional[int] = graph_kwargs.get("graph_id", None)
-    cpp_wrapper: bool = graph_kwargs.get("cpp_wrapper", False)
-    aot_mode: bool = graph_kwargs.get("aot_mode", False)
-    is_inference: bool = graph_kwargs.get("is_inference", False)
-    layout_opt: Optional[bool] = graph_kwargs.get("layout_opt", None)
-    extern_node_serializer: Optional[
-        Callable[[List[ExternKernelNode]], Any]
-    ] = graph_kwargs.get("extern_node_serializer", None)
-    boxed_forward_device_index: Optional[BoxedDeviceIndex] = graph_kwargs.get(
-        "boxed_forward_device_index", None
-    )
+class FxCompile(ABC):
```

Comment:
y'know, I have no idea what ABC is good for LOL

Reply:
It's kind of unnecessary - mostly it just makes sure you don't accidentally instantiate that class.  All the @abstractmethod  stuff still works without it.